src/explorer/WebAppProvider.ts

```diff
@@ -59,15 +57,17 @@ export class WebAppProvider extends SubscriptionTreeItem {
         );
     }
 
-    public async createChildImpl(showCreatingTreeItem: (label: string) => void, actionContext: IActionContext): Promise<AzureTreeItem> {
-        const workspaceConfig: WorkspaceConfiguration = workspace.getConfiguration(extensionPrefix);
-        const advancedCreation: boolean | undefined = workspaceConfig.get(configurationSettings.advancedCreation);
-        const newSite: Site | undefined = await createWebApp(actionContext, this.root, { advancedCreation }, showCreatingTreeItem);
-        if (newSite === undefined) {
-            throw new UserCancelledError();
+    public async createChildImpl(showCreatingTreeItem: (label: string) => void, createOptions?: IAppCreateOptions): Promise<AzureTreeItem> {
+        if (createOptions && createOptions.actionContext) {
+            const newSite: Site | undefined = await createWebApp(createOptions.actionContext, this.root, createOptions, showCreatingTreeItem);
+            if (newSite === undefined) {
+                throw new UserCancelledError();
+            } else {
+                const siteClient: SiteClient = new SiteClient(newSite, this.root);
+                return new WebAppTreeItem(this, siteClient);
+            }
         } else {
-            const siteClient: SiteClient = new SiteClient(newSite, this.root);
-            return new WebAppTreeItem(this, siteClient);
+            throw new Error('Missing object "actionContext"');
```

Comment:
I wasn't really sure what to do in this scenario (but don't expect to run into it either)

Reply:
This is actually pretty common: https://github.com/Microsoft/vscode-azuretools/issues/120

Just create a fake one: https://github.com/Microsoft/vscode-azurefunctions/blob/master/src/tree/FunctionAppProvider.ts#L73